qwiic_i2c/micropython_i2c.py

```diff
@@ -182,3 +192,24 @@ def ping(self, devAddress):
 	def scan(self):
 		""" Returns a list of addresses for the devices connected to the I2C bus."""
 		return self._i2cbus.scan()
+	
+	#-----------------------------------------------------------------------
+	# Custom method for reading +8-bit register using `i2c_msg` from `smbus2`
+	#
+	# Designed to have same operation as the __i2c_rdwr method in linux_i2c.py
+	def __i2c_rdwr__(self, address, write_message, read_nbytes):
```

Comment:
This method should be changed. I know this is what was used elsewhere, but it's not a great name. Should match the conventions of the other methods in this driver, and get added to i2c_driver.py and linux_i2c.py. Then any device drivers that use it (I think just the VL53?) should be changed to use this new method. Does that make sense?

Reply:
The naming convention was based off the original [`i2c_rdwr()`](https://github.com/kplindegaard/smbus2/blob/master/smbus2/smbus2.py#L647) method from the smbus2 Python package that this package utilizes as the driver on Linux platforms:

> ```py
> def i2c_rdwr(self, *i2c_msgs):
> ```

When we added this method, I included the double underscores (`__`) to give it a private attribute. This keeps the method hidden *(in a way)*, since we were only using it for the VL53L1x. Additionally, the method wasn't considered a feature of the package, as noted in the [`README.md` file](https://github.com/kplindegaard/smbus2?tab=readme-ov-file#i2c) of the `smbus2` GitHub repository, :

> [!NOTE]
> Starting with v0.2, the smbus2 library also has support for combined read and write transactions. *`i2c_rdwr`* is not really a SMBus feature but comes in handy when the master needs to:
>
> 1. read or write bulks of data larger than SMBus' 32 bytes limit.
> 1. write some data and then read from the slave with a repeated start and no stop bit between.
>
> Each operation is represented by a *`i2c_msg`* message object.

---

As a recommendation... I'd keep the `__i2c_rdwr__()`method for backwards compatibility. Since it is technically private, we could create a new *public* method with the naming convention that you prefer. To keep the functionality, you could:

- Call on the `__i2c_rdwr__()`method
- Reuse the same code